src/interpreter/index.ts

```diff
@@ -237,15 +234,19 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
         // Set the focused node of the sub env, because our current env has the most up-to-date reference.
         newEnv.setFocusedNode(this._environment.getFocusedNode());
 
+        let returnValue;
         try {
             this._environment = newEnv;
-            return func(this);
+            returnValue = func(this);
+            this._environment = originalEnvironment;
+            this._environment.setFocusedNode(newEnv.getFocusedNode());
         } catch (err) {
-            throw err;
-        } finally {
```

Comment:
it turns out, if you try to `throw` an error in `catch`, but you _also_ have a `finally`, then the `throw` is just ignored. so we were throwing errors but not propagating them upwards.